src/command_transfer.h

```diff
@@ -0,0 +1,63 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "usb/usb-device.h"
+
+#include <vector>
+#include <stdint.h>
+
+namespace librealsense
+{
+    namespace platform
+    {
+        class command_transfer
+        {
+        public:
+            virtual std::vector<uint8_t> send_receive(
+                const std::vector<uint8_t>& data,
+                int timeout_ms = 5000,
+                bool require_response = true) = 0;
+
+            virtual ~command_transfer() = default;
+        };
+
+        class command_transfer_usb : public command_transfer
+        {
+        public:
+            command_transfer_usb(rs_usb_device device) : _device(device) {}
+            ~command_transfer_usb() {};
+
+            std::vector<uint8_t> send_receive(
+                const std::vector<uint8_t>& data,
+                int timeout_ms = 5000,
+                bool require_response = true) override
+            { 
+                const auto& m = _device->open();
+                auto intfs = _device->get_interfaces(USB_SUBCLASS_HWM);
+                if (intfs.size() == 0)
+                    throw std::runtime_error("can't find HWM interface");
+
+                auto hwm = intfs[0];
+
+                int transfered_count = m->bulk_transfer(hwm->first_endpoint(USB_ENDPOINT_DIRECTION_WRITE), const_cast<uint8_t*>(data.data()), data.size(), timeout_ms);
```

Comment:
Pls check whether try/catch for bulk_transfer is needed as well

Reply:
bulk transfer may throw, why do you think we should be catch it here?

Reply:
I changed `bulk_transfer ` to return `usb_status ` and it shouldn't throw